universal.md

```diff
@@ -0,0 +1,423 @@
+3. A few generic object types are supported. Notably these include "Pointer", "Relation" and "Property". Pointer is simply an object that points at another one which should be dereferenced. A signed Pointer object can be used for a detached signature. Relation is a pointer, a type, and a second object which is related, used for adding metadata such as an SBOM to an existing object; again a client that is not interested in the SBOM should just follow the link to the primary object. "Property" is like a pointer but is stored within a manifest and links to a property of that manifest itself, used if you want to say store an SBOM at build time in the same object, to avoid an extra redirection.
+4. We support items in a manifest that have neither blobs nor references, so that the server side does not care about these. This allows manifests to do things like reference foreign layers or images by tag that are not actually references at all, but make the client side processing more consistent.
+5. User interface specific items, such as links to icons for the image to be displayed can be added as an additional object that the client will ignore but can be used by the registry UI only.
+6. The manifest as a whole does not have a type, only the objects in it.
+
+- **`schemaVersion`** *int*
+
+  This is a REQUIRED property. The design of this document is such that this should not need to be bumped.
+
+- **`mediaType`** *string*
+
+  This field contains the `mediaType` of this document. This MUST be `application/vnd.oci.object.manifest.v1+json` for the JSON encoding or `application/vnd.oci.object.manifest.v1+jwt` for JSON web signature (RFC7515) encoding. The JSON web signature encoding allows an inline signature on the object. Other formattings could be allowed in future.
+
+- **`objects`** *array of objects*
+
+  An OPTIONAL list of objects that make up this manifest. These will potentially include multiple objects of the same or differetn versions that the client will pick. Blob and reference links are extracted from this list for the server housekeeping activity. Note we could use a map rather than a list here, with keys corresponding to object `type`.
+
+The `objects` each correspond to items that the client may wish to process, and are defined as follows.
+
+- **`type`** *string*
+
+  A REQUIRED type for this object. Official allocations will be registered with OCI, such as for container images (eg `org.oci.image`), and other allocations will be recommended to be registered as standard. These should be reverse DNS allocations to avoid conflicts.
+
+- **`version`** *string*
```

Comment:
Currently, versions are embedded in the `mediaType`. I kind of like having the `type` and `version` be separate from a client's perspective, but it's unclear to me that breaking from the existing model is worth the benefit.

This also is subject to the same self-describing object "code smell" as having the mediaType embedded in a manifest.